packages/theme/src/cli/utilities/theme-environment/assets.ts

```diff
@@ -0,0 +1,42 @@
+import {readFile} from '@shopify/cli-kit/node/fs'
+import {lookupMimeType} from '@shopify/cli-kit/node/mimes'
+import {defineEventHandler, serveStatic} from 'h3'
+import {joinPath} from '@shopify/cli-kit/node/path'
+import {stat} from 'fs/promises'
+
+/** Forces assets to be served from local domain */
+export function replaceLocalAssets(html: string) {
```

Comment:
In order to reduce the dependency on the CDN URL structure, I believe we may use the `DevServerContext.localThemeFileSystem` to assist us.

Let's say we have `assets/base.css`, `assets/reset.css`, `assets/panel.js`, and `assets/panel.css.liquid` in the theme, and let's say that we built a regular expression that captures the URLs of all script and link tags on the page (this regex really just captures all URLs used in link and script tags, regardless of the domain). Then we get this:

```
[
  "https://store.myshopify.com/cdn/shop/t/28/assets/base.css?v=149927614001461873041712915543",
  "https://store.myshopify.com/cdn/shop/t/28/assets/reset.css?v=149927614001461873041712915232",
  "https://store.myshopify.com/cdn/shop/t/28/assets/panel.js?v=149927614001461873041712915231",
  "https://store.myshopify.com/cdn/shop/t/28/assets/panel.css?v=149927614001461873041712910984",
  "https://store.myshopify.com/cdn/shop/t/28/compiled_assets/scripts.js?7538",
  ...
]
```

Then, by using the JS/CSS files (`assets/base.css`, `assets/reset.css`, `assets/panel.js`) from the local file system, we may infer which URLs must be replaced:

```
"https://store.myshopify.com/cdn/shop/t/28/assets/base.css?v=149927614001461873041712915543",
"https://store.myshopify.com/cdn/shop/t/28/assets/reset.css?v=149927614001461873041712915232",
"https://store.myshopify.com/cdn/shop/t/28/assets/panel.js?v=149927614001461873041712915231",
```

Finally, we replace them with:
```
"/assets/base.css?v=149927614001461873041712915543",
"/assets/reset.css?v=149927614001461873041712915232",
"/assets/panel.js?v=149927614001461873041712915231",
```

This process is worthwhile because:
- We need to avoid replacing all URLs (as `https://store.myshopify.com/cdn/shop/t/28/assets/panel.css?v=149927614001461873041712910984`/`assets/panel.css.liquid` cannot be served locally and must be rendered/served remotely).
- We avoid coupling our regular expression with Shopify CDN structure that might change. So, if it changes to something like `https://store.myshopify.com/shopify_cdn/shop/t/28/assets/base.css`, this module will still work.